expression/generator/time_vec.go

```diff
@@ -464,14 +569,60 @@ func (g gener) gen() interface{} {
 	return result
 }
 
+{{ define "datetimeGener" }}
+	{{- if eq .TypeA.ETName "String"}}
+		&dateStrGener{NullRation: 0.2},
+	{{- else if eq .TypeA.ETName "Int"}}
+		&dateTimeIntGener{nullRation: 0.2},
+	{{- else }}
+		&defaultGener{eType: types.ET{{.TypeA.ETName}}, nullRation: 0.2},
+	{{- end }}
+{{ end }}
+
+{{ define "intervalGener" }}
+	{{- if eq .TypeB.ETName "String" -}}
+		&numStrGener{rangeInt64Gener{math.MinInt32 + 1, math.MaxInt32}},
+	{{- else -}}
+		&defaultGener{eType: types.ET{{.TypeB.ETName}}, nullRation: 0.2},
+	{{- end }}
+{{ end }}
+
+{{ define "addOrSubDateCases" }}
+	{{ range .Sigs }} // {{ .SigName }}
+		{
+			retEvalType: types.ET{{ .Output.ETName }},
+			{{- if eq .TestTypeA "" }}
+			childrenTypes: []types.EvalType{types.ET{{ .TypeA.ETName }}, types.ET{{ .TypeB.ETName }}, types.ETString},
+			{{- else }}
+			childrenTypes: []types.EvalType{types.ET{{ .TestTypeA }}, types.ET{{ .TestTypeB }}, types.ETString},
+			{{- end }}
+			{{- if ne .FieldTypeA "" }}
+			childrenFieldTypes: []*types.FieldType{types.NewFieldType(mysql.Type{{.FieldTypeA}}), types.NewFieldType(mysql.Type{{.FieldTypeB}})},
+			{{- end }}
+			geners: []dataGenerator{
+				{{- if eq .TestTypeA "" }}
+					{{- template "datetimeGener" . -}}
+					{{- template "intervalGener" . -}}
+					nil,
+				{{- else }}
+					{{- template "datetimeGener" . -}}
+					{{- template "intervalGener" . -}}
+					nil,
+				{{- end }}
```

Comment:
This `if` branch seems unnecessary?
Rest LGTM.

Reply:
Deleted.